sync-core/src/main/java/com/cloudant/sync/replication/BasicPushStrategy.java

```diff
@@ -313,7 +316,7 @@ private ItemsToPush missingRevisionsToJsonDocs(
         return allOpenRevisions;
     }
 
-    public String getReplicationId() {
+    public String getReplicationId() throws DatastoreException {
```

Comment:
Throwing this exception appears to terminate the replication, whereas in the pull case it just causes the putting of the checkpoint doc to fail and an error is logged, but the replication continues. Should the push also just log a warning if the checkpointing fails?